apps/meteor/ee/server/lib/EnterpriseCheck.ts

```diff
@@ -0,0 +1,58 @@
+import type { ServiceSchema } from 'moleculer';
+
+const { LICENSE_CHECK_INTERVAL = '20', MAX_FAILS = '2' } = process.env;
+
+let checkFails = 0;
+const maxFails = parseInt(MAX_FAILS) || 2;
+
+export const EnterpriseCheck: ServiceSchema = {
+	name: 'EnterpriseCheck',
+	methods: {
+		/**
+		 * The main idea is if there is no scalability module enabled,
+		 * then we should not allow more than one service per environment.
+		 * So we list the services and nodes, and if there is more than
+		 * one, we say it should be shutdown.
+		 */
+		async shouldShutdown(): Promise<boolean> {
+			const services: {
+				name: string;
+				nodes: string[];
+			}[] = await this.broker.call('$node.services');
+
+			const currentService = services.find((service) => {
+				return service.name === this.name;
+			});
+
+			// if current service is not on the list maybe it is already shut down?
+			if (!currentService) {
+				return true;
+			}
+
+			const { nodes } = currentService;
+
+			const firstNode = nodes.sort().shift();
+
+			// if the first node is the current node and there are others nodes running the same service, then it should shutdown
+			return firstNode === this.broker.nodeID && nodes.length > 0;
```

Comment:
So in rare cases.. this might depend on outside orchestrator to pop a node back online?

Reply:
no, that's what the `sort` tries to solve.. if 2 nodes online performing the check at the same time, the "first node" will be the same for both, that means only one of them will shut down.

Reply:
Got it!  That makes sense 👍